src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -808,5 +822,67 @@ private int NodeIndexForContext(BuildEventContext context)
             : path;
     }
 
+    internal static string FormatEventMessage(
+            string category,
+            string subcategory,
+            string? message,
+            string code,
+            string? file,
+            int lineNumber,
+            int endLineNumber,
+            int columnNumber,
+            int endColumnNumber)
+    {
+        using SpanBasedStringBuilder builder = new(128);
+
+        if (string.IsNullOrEmpty(file))
+        {
+            builder.Append("MSBUILD : ");    // Should not be localized.
```

Comment:
Should we update this prefix/formatting to be more modern?

Reply:
Agree with this comment, but I'm not sure what I would put here instead. Also, as @rainersigwald mentioned in another comment, if we change this prefix we may invalidate existing problem matchers, right?

Reply:
For reference, the existing VSCode problem matcher is here: https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/tasks/common/problemMatcher.ts#L1286-L1294

Reply:
I don't know of anything that currently matches the fallback `MSBUILD : ` prefix, but yeah looking at that we probably have to put _some_ prefix there where the path would go.